elasticsearch/run.sh

```diff
@@ -60,4 +69,72 @@ else
 	exit 1
 fi
 
+
+# Wait for Elasticsearch port to be opened. Fail on timeout or if response from Elasticsearch is unexpected.
+wait_for_port_open() {
+	rm -f ${LOG_FILE}
+	echo -n "Checking if Elasticsearch is ready on ${ES_REST_BASEURL} "
+	while ! curl -s -k \
+			--cert ${secret_dir}admin-cert \
+			--key  ${secret_dir}admin-key \
+			--max-time ${max_time} \
+			-o ${LOG_FILE} \
+			${ES_REST_BASEURL} && [ ${timeouted} == false ]
+	do
+		echo -n "."
+		sleep ${RETRY_INTERVAL}
+		(( retry -= 1 ))
+		if (( retry == 0 )) ; then
+			timeouted=true
+		fi
+	done
+
+	# Test for response code 200 in Elasticsearch output. This can be sensitive to Elasticsearch version.
+	if [ -f ${LOG_FILE} ] && grep -q "200" ${LOG_FILE} ; then
+		echo "- connection successful"
+	else
+		if [ ${timeouted} == true ] ; then
+			echo -n "[timeout] "
+		fi
+		echo "failed"
+		cat ${LOG_FILE}
+		exit 1
+	fi
+}
+
+verify_or_add_index_templates() {
+	wait_for_port_open
+	# Try to wait for cluster become more stable before index template being pushed in.
+	# Give up on timeout and continue...
+	curl -v -s -k -X GET \
+		--cert ${secret_dir}admin-cert \
+		--key  ${secret_dir}admin-key \
+		"${ES_REST_BASEURL}/_cluster/health?wait_for_status=yellow&timeout=${max_time}s"
```

Comment:
I ran into a similar [inner] debate for the openshift-elasticsearch-plugin which does the ACL seeding for searchguard. Based on scenarios seen for a customer (where their cluster was taking a long time to recover and some indices had even failed to recover). I would still want the ACLs in place so they could go in as an admin and clean up/recover manually if need be.

In that case I changed my health check from a waitforyellow to just getting a response back from the cluster health api.

I think we would want to consider something similar here. If we haven't yet created the index templates, we would want to make sure we seed them as soon as the cluster is responsive in case a customer begins to send logs to it (via Fluentd) because they see that ES has completed loading its indices in. The cluster may be in a 'red' state at this time because it hadn't finished recovering but the customer may not be aware of this.  And we would want to make sure that the index templates were there before any new indices are created.

I think regarding the individual vs cluster claim isn't an issue if we're waiting for the cluster/node to be responsive. A non-master node would be checking to see if its going to be master before it starts to do recovery I thought. If so, I think it would be connected to the cluster before it was responsive to health checks.